### ISSUE TITLE
What license applies to the code in this repo?

### ISSUE THREAD

**Issue by jayrulez** [cocos/engine-native#3971]
The readme suggests that the code is MIT:
![image](https://user-images.githubusercontent.com/78807/137025104-d625c1b9-6d0d-489b-a008-340781639112.png)

However, the license text in the source code files states something different.
```
/****************************************************************************
 Copyright (c) 2019-2021 Xiamen Yaji Software Co., Ltd.
 http://www.cocos.com
 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated engine source code (the "Software"), a limited,
 worldwide, royalty-free, non-assignable, revocable and non-exclusive license
 to use Cocos Creator solely to develop games on your target platforms. You shall
 not use Cocos Creator software for developing other software or tools that's
 used for developing games. You are not granted to publish, distribute,
 sublicense, and/or sell copies of Cocos Creator.
 The software or tools in this License Agreement are licensed, not sold.
 Xiamen Yaji Software Co., Ltd. reserves all rights not expressly granted to you.
 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
****************************************************************************/
```

Could you please clarify which one is valid?

**Comment by pandamicro** [comment]
It's not MIT, it's modified version of MIT and it's not allowed to use the engine code to create other game engine tools

**Comment by pandamicro** [comment]
> The readme suggests that the code is MIT:

I also updated the readme, now it's pointing to the correct license file located in [https://github.com/cocos/cocos-engine/] repo

**Comment by Aspie96** [comment]
(I am not a lawyer and this is not legal advice).

> It's not MIT, it's modified version of MIT and it's not allowed to use the engine code to create other game engine tools
> […]
> I also updated the readme, now it's pointing to the correct license file located in the latest [engine](https://github.com/cocos/cocos-engine/) repo.

@pandamicro, why is it that the LICENSE file in the new repo is the MIT license?

Furthermore, the MIT license plainly contradicts the following:

https://github.com/cocos/cocos-engine/blob/develop/licenses/EDITOR_license_and_terms_en.txt

> Open sourced JavaScript and C++ game engine, with editor extension can be modified and customized by developer user. The modified and customized version of software can be used to develop game software and interactive rich media for end users, but **not allowed to redistribute or publish game development software based on Cocos Creator's open sourced code**.

The MIT license (the one in the root of the repo) absolutely does allow one to redistribute or publish game development software based on the open source code.

It's also part of the [OSD](https://opensource.org/osd/) that code with such a restriction makes a license not open source (although it can be source available).

It seems to me that there either is an error in the root license of the project, which should have been something other than MIT, or the "EDITOR_license_and_terms_en.txt" file is out-dated. If it's out-dated, and the intended license prohibits use of the engine code to create other game engine tools, then it really shouldn't be referred to as "open source", there or elsewhere.

**Comment by pandamicro** [comment]
Sorry for losting track of this, I will outline the most important conclusion and then show you guys the timeline of changes.

Conclusion: **Cocos engine used by Cocos Creator is now unified into one repo [cocos/cocos-engine](https://github.com/cocos/cocos-engine/), and this repo is now fully open source under MIT license.** The editor's agreement is put into the repo as a reference, which should not limit the usage of the open source repo as its own.

Cocos has really long history, Cocos Creator also have multiple major versions and we tried our best to deliver the awesome tool with a total open source engine. Here are some events listed in timeline to explain things better:

- 06 May 2022: The Engine Native repo is merged into the cocos/cocos-engine repo in v3.5, reference in [the release note here](https://discuss.cocos2d-x.org/t/cocos-creator-v3-5-released/55944#engine-repository-transfer-and-customization-13)
- 28 Dec 2022: The cocos/cocos-engine repo have updated its license to MIT: https://github.com/cocos/cocos-engine/pull/13927
- A root license have been updated the same day: https://github.com/cocos/cocos-engine/commit/9e7169f19ae73a083626eb667de8b8e43990fcc5

Finally, as this repo is no longer tracked, I will close this issue
